src/metabase/formatter/datetime.clj

```diff
@@ -65,29 +65,21 @@
         (all-cols-settings {::mb.viz/column-name (or field-id-or-name column-name)}))))
 
 (defn- determine-time-format
-  "Given viz-settings with a time-style and possible time-enabled (precision) entry, create the format string."
-  [{:keys [time-enabled time-style] :or {time-enabled "minutes" time-style "h:mm A"}}]
-  (let [base-time-format (case time-enabled
-                           "minutes" "mm"
-
-                           "seconds" "mm:ss"
-
-                           "milliseconds" "mm:ss.SSS"
-
-                           ;; {::mb.viz/time-enabled nil} indicates that time is explicitly disabled, rather than
-                           ;; defaulting to "minutes"
-                           nil nil)]
-    (if base-time-format
-      (case time-style
-        "HH:mm" (format "HH:%s" base-time-format)
-
-        ;; Deprecated time style which should be already converted to HH:mm when viz settings are
-        ;; normalized, but we'll handle it here too just in case. (#18112)
-        "k:mm" (str "h" base-time-format)
-
-        ("h:mm A" "h:mm a") (format "h:%s a" base-time-format)
-
-        time-style)
+  "Given viz-settings with a time-style and possible time-enabled (precision) entry, create the format string.
+  Note that if the `:time-enabled` key is present but the value is nil, we explicitly do not show the time."
+  [{:keys [time-style] :or {time-style "h:mm A"} :as viz-settings}]
+  ;; NOTE - If :time-enabled is present but nil it will return nil
+  (when-some [base-time-format (case (get viz-settings :time-enabled "minutes")
+                               "minutes" "mm"
+                               "seconds" "mm:ss"
+                               "milliseconds" "mm:ss.SSS"
+                               nil nil)]
+    (case time-style
+      "HH:mm" (format "HH:%s" base-time-format)
+      ;; Deprecated time style which should be already converted to HH:mm when viz settings are
+      ;; normalized, but we'll handle it here too just in case. (#18112)
```

Comment:
Is there some way we can ensure this is deprecated appropriately and eliminate this part?

Reply:
Good idea. Probably something to follow up on. This was lifted from some code @noahmoss wrote in the xlsx ns. We should all discuss.